zkevm-circuits/src/circuit_tools/constraint_builder.rs

```diff
@@ -29,19 +29,92 @@ fn get_condition_expr<F: Field>(conditions: &Vec<Expression<F>>) -> Expression<F
 
 /// Data for dynamic lookup
 #[derive(Clone, Debug)]
-pub struct DynamicData<F> {
+pub struct LookupData<F> {
     /// Desciption
     pub description: &'static str,
     /// Condition under which the lookup needs to be done
-    pub condition: Expression<F>,
+    pub regional_condition: Expression<F>,
```

Comment:
My idea is:
- `local_condition`: conditions that you get directly from constraint builder locally (when you're within a region, all conditions are the from ifx! and matchx! within that region), and I apply those condition directly in `add_lookup` so no need to worry about them during region transitions.
- `regional_condition`: conditions that are left out prior to entering this region, these are the ones that you temper with in `pop_region()`.
I think this separation would be less prone to error, since you know they gave me so much headache last time.

Reply:
The rule of thumb is you don't touch regional condition when you're inside a region, I'll make it clear in code comments.